embulk-core/src/main/java/org/embulk/jruby/JRubyClassLoader.java

```diff
@@ -0,0 +1,120 @@
+    protected void addURL(final URL url) {
+        throw new UnsupportedOperationException("JRubyClassLoader does not support addURL.");
+    }
+
+    @Override
+    public void close() throws IOException {
+        super.close();
+    }
+
+    @Override
+    public URL[] getURLs() {
+        return super.getURLs();
+    }
+
+    /**
+     * Loads the class with the specified binary name from JRuby.
+     *
+     * <p>It prioritizes a class found by this class loader over a class loaded by the parent class loader.
+     */
+    @Override
+    protected Class<?> loadClass(final String name, final boolean resolve) throws ClassNotFoundException {
+        synchronized (this.getClassLoadingLock(name)) {
+            // If a class of the specified name has already been loaded by this class loader, or the parent class loader,
+            // find the loaded class, and return it.
+            final Class<?> loadedClass = this.findLoadedClass(name);
+            if (loadedClass != null) {
+                return this.resolveClassIfNeeded(loadedClass, resolve);
+            }
+
+            // JRuby should use Joda-Time of embulk-core (on the top-level class loader), not of jruby-complete.
+            // Otherwise, embulk-core uses its own, and JRuby uses its own, then they wouldn't match.
+            //
+            // TODO: Remove the condition when embulk-core removes Joda-Time from its dependencies.
+            if (!name.startsWith("org.joda.time.")) {
+                // If a class of the specified name has not been loaded yet, and is found by this (not parent) class loader,
+                // find it, and return it.
+                try {
+                    return this.resolveClassIfNeeded(this.findClass(name), resolve);
+                } catch (final ClassNotFoundException ignored) {
+                    // Passing through intentionally.
+                }
+            }
+
+            // If a class of the specified name is found by this class loader (not by the parent class loader),
+            // find it, and return it.
+            try {
+                return this.resolveClassIfNeeded(this.getParent().loadClass(name), resolve);
+            } catch (final ClassNotFoundException ignored) {
+                // Passing through intentionally.
+            }
+
+            throw new ClassNotFoundException(name);
+        }
+    }
+
+    /**
+     * Finds the resource with the given name from JRuby.
+     *
+     * <p>It prioritizes a resource found by this class loader over a class loaded by the parent class loader.
+     */
+    @Override
+    public URL getResource(final String name) {
+        final URL parentUrl = this.getParent().getResource(name);
+        if (parentUrl != null) {
+            return parentUrl;
+        }
+
+        return this.findResource(name);
+    }
+
+    @Override
+    public Enumeration<URL> getResources(final String name) throws IOException {
+        final Vector<URL> resources = new Vector<URL>();
```

Comment:
nit: can be `final Vector<URL> resources = new Vector<>();`

Reply:
Fixed in ef7af455559cec2eaf52ed5541bc503de7ae3cf0.